libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -156,7 +156,7 @@ void ArrowAdapter::release_array(struct ArrowArray* array) {
         // Dictionary arrays are allocated differently than data arrays
         // We need to free the buffers one at a time, then we can call the
         // release schema to continue the cleanup properly
-        for (size_t i = 0; i < array->dictionary->n_buffers; ++i) {
+        for (int64_t i = 0; i < array->dictionary->n_buffers; ++i) {
```

Comment:
This was from a compiler warning I found. Note that `n_children` is of type `int64_t` for both `ArrowSchema` and `ArrowArray` within `nanoarrow.h`.